osutil/user.go

```diff
@@ -137,6 +137,40 @@ func AddUser(name string, opts *AddUserOptions) error {
 	return nil
 }
 
+type UserDelOptions struct {
+	Force     bool
+	Remove    bool
+	ExtraUser bool
+}
+
+func UserDel(name string, opts *UserDelOptions) error {
+	_, err := userLookup(name)
+	if err != nil {
+		return fmt.Errorf("cannot find user %q: %s", name, err)
```

Comment:
I don't have a strong feeling about this, just wondering if we should be relaying this to `userdel` instead doing an early check here.

Reply:
Yeah, there is also the TOCTOU issue, i.e. this is inherently racy (not sure if userdel is any better internally though, i.e. if it locks but I suspect it does :)

Reply:
Agree that the test is redundant, userdel should handle this. It may or may not be race-free inside, but it's a black box for us and we don't control it, so let's fix what we have control over.

Reply:
I'm not sure this extra check helps  (we are already doing this kind of check in the caller)

Reply:
Removed the extra check. Also it seems that this test attracted our attention there and we all missed a real bug: the file under sudoers.d for that user was not being removed. Fixing that too.